crates/fullnode/src/runner.rs

```diff
@@ -320,6 +323,18 @@ where
             }
         };
 
+        if let Some(config) = &self.pruning_config {
+            let pruner = Pruner::<DB>::new(
+                config.clone(),
+                self.ledger_db.get_last_pruned_l2_height()?.unwrap_or(0),
+                self.soft_confirmation_tx.subscribe(),
+                self.ledger_db.clone(),
+            );
+
+            self.task_manager
+                .spawn(|cancellation_token| pruner.run(cancellation_token));
```

Comment:
This is not a blocking task. Is it ok?

Reply:
Yes that's okay... i am `spawn_blocking` inside the pruner.